plugins/by-name/oil-git-status/default.nix

```
@@ -49,29 +58,53 @@ lib.nixvim.plugins.mkNeovimPlugin {
           ${options.plugins.oil.enable} = true;
         '';
       }
-      {
-        when =
-          let
-            value = config.plugins.oil.settings.win_options.signcolumn or null;
-            isKnownBadStr = lib.lists.elem value [
-              "yes"
-              "yes:"
-              "yes:0"
-              "yes:1"
-            ];
-            hasYes = lib.strings.hasInfix "yes" value;
-          in
-          !(lib.strings.isString value && hasYes && !isKnownBadStr);
+      (
+        # The plugin requires the oil configuration allows at least 2 sign columns.
+        # They suggest `win_options.signcolumn = "yes:2"`
+        # See also `:h 'signcolumn'`
+        let
+          # Get `signcolumn` setting value
+          value = config.plugins.oil.settings.win_options.signcolumn or null;
+          isStr = builtins.isString value;
 
-        message = ''
-          This plugin requires the following `plugins.oil` setting:
-            ${options.plugins.oil.settings} = {
-              win_options = {
-                signcolumn = "yes:2";
-              };
-            };`
-        '';
-      }
+          # Split parts
+          parts = lib.strings.splitString ":" value;
+          left = builtins.elemAt parts 0;
+          right = builtins.elemAt parts 1;
+          hasRight = builtins.length parts > 1;
+
+          # Check left
+          validLeft = builtins.elem left [
+            "auto"
+            "yes"
+          ]; # TODO: is "auto:2" valid?
```

Can we confirm whether `"auto"` is a valid left-hand-side compatible with oil-git-status? I'd like to drop the TODO, ideally before merging.

I guess we'd also want to know whether the right-hand-side number has different semantics for `"auto"` vs `"yes"`

If `"auto"` isn't valid, we can drop it:
```suggestion
          # Check left
          validLeft = left == "yes";
```